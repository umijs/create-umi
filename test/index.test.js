const coffee = require('coffee');
const path = require('path');
const os = require('os');
const fs = require('fs');
const { winEOL } = require('umi-utils');

describe('test umi-create', () => {
  beforeAll(() => {
    process.env.TEST = 'test';
  });
  it('test generate antd pro project from github', async () => {
    let temp = fs.mkdtempSync(path.join(os.tmpdir(), `umi-create`));
    if (os.platform() === 'darwin') {
      temp = path.join('/private', temp);
    }
    const Coffee = coffee.Coffee;
    const fixtures = path.join(__dirname, 'fixtures');
    const response = await new Coffee({
      method: 'fork',
      cmd: path.join(__dirname, '../cli.js'),
      opt: { cwd: temp, stdout: process.stdout, stderr: process.stderr, stdin: process.stdin },
    })
      .beforeScript(path.join(fixtures, 'mock_github.js'))
      .waitForPrompt()
      .write('\n')
      .writeKey('DOWN', 'ENTER')
      .write('\n')
      .end();

    expect(winEOL(response.stdout).replace(/>/g, '❯')).toMatchSnapshot();
    expect(response.code).toBe(0);
    expect(fs.existsSync(path.join(temp, 'package.json'))).toBeTruthy();
  });

  it('test generate antd pro project from gitee.org', async () => {
    let temp = fs.mkdtempSync(path.join(os.tmpdir(), `umi-create-cnpm`));
    if (os.platform() === 'darwin') {
      temp = path.join('/private', temp);
    }
    const Coffee = coffee.Coffee;
    const fixtures = path.join(__dirname, 'fixtures');
    const response = await new Coffee({
      method: 'fork',
      cmd: path.join(__dirname, '../cli.js'),
      opt: { cwd: temp },
    })
      .beforeScript(path.join(fixtures, 'mock_cnpmjs.js'))
      .waitForPrompt()
      .write('\n')
      .write('\n')
      .write('\n')
      .end();

    expect(winEOL(response.stdout).replace(/>/g, '❯')).toMatchSnapshot();
    expect(response.code).toBe(0);
    expect(fs.existsSync(path.join(temp, 'package.json'))).toBeTruthy();
  });
});

describe('typescript', () => {
  it('tsconfig.json should be removed if JavaScript is picked', async () => {
    let temp = fs.mkdtempSync(path.join(os.tmpdir(), `umi-create`));
    if (os.platform() === 'darwin') {
      temp = path.join('/private', temp);
    }
    const Coffee = coffee.Coffee;
    const fixtures = path.join(__dirname, 'fixtures');
    const response = await new Coffee({
      method: 'fork',
      cmd: path.join(__dirname, '../cli.js'),
      opt: { cwd: temp },
    })
      .beforeScript(path.join(fixtures, 'mock_github.js'))
      .waitForPrompt()
      .write('\n')
      .write('\n')
      .write('\n')
      .end();

    expect(response.code).toBe(0);
    expect(fs.existsSync(path.join(temp, 'jsconfig.json'))).toBeTruthy();
    expect(fs.existsSync(path.join(temp, 'tsconfig.json'))).toBeFalsy();
    expect(fs.existsSync(path.join(temp, 'package.json'))).toBeTruthy();
  });
});
