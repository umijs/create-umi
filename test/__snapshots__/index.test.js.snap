// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test umi-create test generate antd pro project from gitee.org 1`] = `
"? Select the boilerplate type (Use arrow keys)
❯ ant-design-pro  - Create project with a layout-only ant-design-pro boilerplate
, use together with umi block. 
  app             - Create project with a simple boilerplate, support typescript
. 
  block           - Create a umi block. 
  library         - Create a library with umi. 
  plugin          - Create a umi plugin. [41D[41C[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G? Select the boilerplate type ant-design-pro[44D[44C
? 🤓 Which language do you want to use? (Use arrow keys)
❯ TypeScript 
  JavaScript [13D[13C[2K[1A[2K[1A[2K[G? 🤓 Which language do you want to use? TypeScript[49D[50C
? 🦄 Time to use better, faster and latest antd@4! (Y/n) [56D[57C[2K[G? 🦄 Time to use better, faster and latest antd@4! Yes[53D[54C
❯   
❯ 使用 antd 4.0 可能会有一些兼容性问题，阅读下面的文档了解具体的改动
❯ There may be some compatibility issues when using antd 4.0. Read the following documents for specific changes
❯ https://next.ant.design/docs/react/migration-v4-cn
❯   
❯ 🚚 clone success
❯ [Sylvanas] Prepare js environment...
❯ [JS] Clean up...
❯ Clean up...
✨ File Generate Done
"
`;

exports[`test umi-create test generate antd pro project from github 1`] = `
"? Select the boilerplate type (Use arrow keys)
❯ ant-design-pro  - Create project with a layout-only ant-design-pro boilerplate
, use together with umi block. 
  app             - Create project with a simple boilerplate, support typescript
. 
  block           - Create a umi block. 
  library         - Create a library with umi. 
  plugin          - Create a umi plugin. [41D[41C[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G? Select the boilerplate type ant-design-pro[44D[44C
? 🤓 Which language do you want to use? (Use arrow keys)
❯ TypeScript 
  JavaScript [13D[13C[2K[1A[2K[1A[2K[G? 🤓 Which language do you want to use? 
  TypeScript 
❯ JavaScript [13D[13C[2K[1A[2K[1A[2K[G? 🤓 Which language do you want to use? JavaScript[49D[50C
? 🦄 Time to use better, faster and latest antd@4! (Y/n) [56D[57C[2K[G? 🦄 Time to use better, faster and latest antd@4! Yes[53D[54C
❯   
❯ 使用 antd 4.0 可能会有一些兼容性问题，阅读下面的文档了解具体的改动
❯ There may be some compatibility issues when using antd 4.0. Read the following documents for specific changes
❯ https://next.ant.design/docs/react/migration-v4-cn
❯   
❯ 🚚 clone success
❯ [Sylvanas] Prepare js environment...
❯ [JS] Clean up...
❯ Clean up...
✨ File Generate Done
"
`;
